nats-experimental/tests/client_tests.rs

```diff
@@ -0,0 +1,48 @@
+// Copyright 2020-2022 The NATS Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+mod nats_server;
+
+mod client {
```

Comment:
everything is `client` in our tests, isn't it?
Not sure why to explicitly embed it.

Reply:
Integration tests so we're testing what's public.
I'm thinking we'll have a few different scopes here
- client
- subscriber
- jetstream::client
- jetstream::subscriber
and so on.

The point of segmenting is to have the ability easily be able to filter with `cargo test` so for example `cargo test client` for all client tests. Then `cargo test subscriber` would test both `client::subscriber` and `jetstream::subscriber`.